sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -665,9 +665,13 @@ void ProgramManager::populateSpecConstRegistry() {
 RTDeviceBinaryImage &ProgramManager::getDeviceImage(OSModuleHandle M,
                                                     KernelSetId KSId,
                                                     const context &Context) {
-  if (DbgProgMgr > 0)
+  if (DbgProgMgr > 0) {
     std::cerr << ">>> ProgramManager::getDeviceImage(" << M << ", \"" << KSId
               << "\", " << getRawSyclObjImpl(Context) << ")\n";
+
+    std::cerr << "available device images:\n";
```

Comment:
Could help when filtering the output...

```suggestion
    std::cerr << "ProgramManager: Available device images:\n";
```

Reply:
actually i just moved this part up because it might be unreachable at it's old place.

If i were to improve those messages i would query for PI_TRACE at runtime instead? what do you think?

Reply:
That would be even better

Reply:
@kbobrovs should i replace all `DbgProgMgr` checks with pi_trace in programmmanager?

Reply:
@hiaselhans , sorry for delay. Yes, definitely makes sense. Can be done as a separate PR.